File: python3/sdk/image_defog_demo.py
# -*- coding:utf-8 -*-

from ais_sdk.gettoken import get_token
from ais_sdk.utils import encode_to_base64
from ais_sdk.utils import decode_to_wave_file
from ais_sdk.image_defog import image_defog
from ais_sdk.image_defog import image_defog_aksk
import json

if __name__ == '__main__':
    #
    # access image defog,post data by token
    #
    user_name = '*****'
    password = '******'
    account_name = '*****'  # the same as user_name in commonly use

    token = get_token(user_name, password, account_name)

    # call interface use base64 file
    result = image_defog(token, encode_to_base64('data/defog-demo-1.png'), '1.5')
    print(result)
    result_obj = json.loads(result)
    decode_to_wave_file(result_obj['result'], 'data/defog-demo_first.png')

    #
    # access image defog,post data by ak,sk
    #
    app_key = "*************"
    app_secret = "************"

    # call interface use base64 file
    result = image_defog_aksk(app_key, app_secret, encode_to_base64('data/defog-demo-1.png'), '1.5')
    print(result)
    result_obj = json.loads(result)
    decode_to_wave_file(result_obj['result'], 'data/defog-demo_second.png')

File: javascript/image_defog_demo.js
/**
 * 图像去雾服务的使用示例
 */
var defog = require("./ais_sdk/image_defog");
var token = require("./ais_sdk/gettoken");
var utils = require("./ais_sdk/utils");

var username = "*******";       // 配置用户名
var domainname = "*******";     // 配置用户名
var password = "*******";       // 密码
var regionName = "cn-north-1";  // 配置区域信息

var filepath = "./data/defog-demo.png";
var data = utils.changeFileToBase64(filepath);
/**
 * token 方式获取结果
 * @type {string}
 */
token.getToken(username, domainname, password, regionName, function (token) {

    defog.defog(token, data, 1.5, true, function (result) {
        var resultObj = JSON.parse(result);
        utils.getFileByBase64Str("./data/defog-demo-token.png", resultObj.result);
    })
});

/**
 * aksk 方式获取结果
 * @type {string}
 */
var app_key = "*************";
var app_secret = "************";

defog.defog_aksk(app_key, app_secret, data, 1.5, true, function (result) {
    var resultObj = JSON.parse(result);
    utils.getFileByBase64Str("./data/defog-demo-aksk.png", resultObj.result);
});

